### ISSUE TITLE
Tensorflow shuffle leak memory

### ISSUE THREAD

**Issue by Yurushia1998** [tensorflow/tensorflow#50406]
My tensorflow pipeline use shuffle and it always end up with OOM. I try to follow solution at https://github.com/tensorflow/tensorflow/issues/44176 but it did not work for me. My tensorflow version is 2.3.0. ANyone know how to fix this?

**Comment by saikumarchalla** [comment]
@Yurushia1998  Could you please provide the simple standalone code/ colab link to reproduce the issue at our end.Thanks

**Comment by Yurushia1998** [comment]
> @Yurushia1998 Could you please provide the simple standalone code/ colab link to reproduce the issue at our end.Also, Please test in TF 2.5 Version as well and let us know.

Hi saikumarchlla, it's a bit hard to provide the full code since it's our ongoing project.
Below is what my input pipeline looks like:
train_holder,label_holder = tf.placeholder(tf.float32, [None, 32, 32, 3],name = “train_holder”), tf.placeholder(tf.int32, [None],name = “label_holder”)
input_tuple = (self.train_holder, self.label_holder)
ds = tf.data.Dataset.from_tensor_slices(data)
map_fn = lambda x, y: (cifar_process(x, is_train), y)
train_dataflow = ds.map(map_fn, num_parallel_calls=tf.data.experimental.AUTOTUNE)

train_ds = train_dataflow.repeat().batch(
self.batch_size, drop_remainder=True
).map(
autoaug_batch_process_map_fn,
num_parallel_calls=tf.data.experimental.AUTOTUNE).prefetch(
buffer_size=tf.data.experimental.AUTOTUNE)
train_input_iterator = (
self.strategy.experimental_distribute_dataset(
train_ds).make_initializable_iterator())

Each time I reinitialize this pipeline with new input by feedict:
self.sess.run([train_input_iterator.initializer],feed_dict = …)


My dataset is has shape (5000,32,32,3), feeded to train_holder, label is a list of shape (50000,)
I need to run self.sess.run([train_input_iterator.initializer],feed_dict = …) many times during training,
Sometimes the memory increase rapidly and I get OOM at this exact step, so I know it's wrong there. This happen only sometimes, and not on my local machine, but when I use a remote cluster with GPU

**Comment by jvishnuvardhan** [comment]
@Yurushia1998 Are you using TF1.x or TF2.x. I am asking it because you are using `tf.placeholder` (which is TF1.x) and mentioned above that you are using `TF2.3` (which is TF2.x). Please note that we are not supporting `TF1.x` anymore.

If you are using TF2.x, can you please try with any open source data or use random data from numpy to show the issue? Also, there were some memory issues with `TF2.3` and `TF2.4` that got resolved in `TF2.5` and `tf-nightly`. If possible use recent TF version and share a standalone code to reproduce the issue. Thanks!
